File: src/main/java/com/now/nowbot/throwable/ModsException.java
package com.now.nowbot.throwable;

public class ModsException extends TipsRuntimeException{
    public enum Type {
        MOD_Receive_CharNotPaired("模组输入异常：字符数量非偶数"),
        MOD_Receive_Conflict("模组输入异常：以下模组冲突：\n%s"),

        ;

        public final String message;

        Type(String message) {
            this.message = message;
        }
    }

    public ModsException(ModsException.Type type){
        super(type.message);
    }

    public ModsException(ModsException.Type type, Object... args){
        super(String.format(type.message, args));
    }
}

File: src/main/java/com/now/nowbot/throwable/botRuntimeException/ModsException.kt
package com.now.nowbot.throwable.botRuntimeException

import com.now.nowbot.throwable.BotException
import com.now.nowbot.throwable.TipsRuntimeException

open class ModsException(message: String) : TipsRuntimeException(message), BotException {
    class CharNotPaired:
        ModsException("模组输入异常：字符数量非偶数")

    class ModConflict(vararg mods: String):
        ModsException("模组异常：以下模组冲突：${mods.joinToString(", ")}")
}
